### PR TITLE
Added management command for initial Open Zaak setup

### PR DESCRIPTION
**Pull request by stevenbal**

### CODE REVIEW

File: src/openzaak/management/commands/setup_configuration.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            )
+            Autorisatie.objects.create(
+                applicatie=notif_api_applicatie_ac,
+                component=ComponentTypes.ac,
+                scopes=[SCOPE_AUTORISATIES_LEZEN],
+            )
+
+            openzaak_applicatie_notif = Applicatie.objects.create(
+                label=f"Open Zaak {municipality.capitalize()}",
+                client_ids=[f"open-zaak-{municipality}"],
```
Comment (stevenbal) [inline src/openzaak/management/commands/setup_configuration.py]: @joeribekker this is step 4 under `Open Zaak` [here](https://open-zaak.readthedocs.io/en/latest/installation/configuration.html). In the manual it specifies that I can use any secret and that it doesn't need to be used in any other step, but for this application (that gives OZ access to notificaties) to work, the clientid/secret must also be registered in Open notificaties right?

Here I used the same clientid/secret as used by the external API credential that for Open Zaak -> Open Notificaties (step #2 in the manual), not sure if that's right

File: src/openzaak/management/commands/setup_configuration.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+from openzaak.components.autorisaties.api.scopes import SCOPE_AUTORISATIES_LEZEN
+
+
+class Command(BaseCommand):
+    help = "Setup the initial necessary configuration"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "openzaak_domain",
+            help="Specifies the domain for this Open Zaak installation",
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: Please provide examples in all help texts, and indicate what its used for.

File: src/openzaak/management/commands/setup_configuration.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            Applicatie.objects.all().delete()
+            openzaak_domain = options["openzaak_domain"]
+            notifications_api_root = options["notifications_api_root"]
+            municipality = options["municipality"]
+            openzaak_to_notif_secret = options["openzaak_to_notif_secret"]
+            notif_to_openzaak_secret = options["notif_to_openzaak_secret"]
+
+            site = Site.objects.get_current()
+            site.domain = openzaak_domain
+            site.name = f"Open Zaak {municipality.capitalize()}"
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: Don't make it capitilized.

File: src/openzaak/management/commands/setup_configuration.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    def handle(self, *args, **options):
+        try:
+            Applicatie.objects.all().delete()
+            openzaak_domain = options["openzaak_domain"]
+            notifications_api_root = options["notifications_api_root"]
+            municipality = options["municipality"]
+            openzaak_to_notif_secret = options["openzaak_to_notif_secret"]
+            notif_to_openzaak_secret = options["notif_to_openzaak_secret"]
+
+            site = Site.objects.get_current()
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py:67]: Please add references to the config docs to all sections.

```suggestion
            # See: https://open-zaak.readthedocs.io/en/latest/installation/configuration.html#setting-the-domain
            site = Site.objects.get_current()
```

File: src/openzaak/management/commands/setup_configuration.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            site.save()
+
+            notif_config = NotificationsConfig.get_solo()
+            notif_config.api_root = notifications_api_root
+            notif_config.save()
+
+            if not APICredential.objects.filter(api_root=notifications_api_root).exists():
+                APICredential.objects.create(
+                    api_root=notifications_api_root,
+                    label=f"Open Notificaties {municipality.capitalize()}",
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: remove cap

File: src/openzaak/management/commands/setup_configuration.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                    client_id=f"open-zaak-{municipality}",
+                    secret=openzaak_to_notif_secret,
+                    user_id=f"open-zaak-{municipality}",
+                    user_representation=f"Open Zaak {municipality.capitalize()}",
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: remove cap

File: src/openzaak/management/commands/setup_configuration.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            notif_config.api_root = notifications_api_root
+            notif_config.save()
+
+            if not APICredential.objects.filter(api_root=notifications_api_root).exists():
+                APICredential.objects.create(
+                    api_root=notifications_api_root,
+                    label=f"Open Notificaties {municipality.capitalize()}",
+                    client_id=f"open-zaak-{municipality}",
+                    secret=openzaak_to_notif_secret,
+                    user_id=f"open-zaak-{municipality}",
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: lower

File: src/openzaak/management/commands/setup_configuration.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+            notif_config = NotificationsConfig.get_solo()
+            notif_config.api_root = notifications_api_root
+            notif_config.save()
+
+            if not APICredential.objects.filter(api_root=notifications_api_root).exists():
+                APICredential.objects.create(
+                    api_root=notifications_api_root,
+                    label=f"Open Notificaties {municipality.capitalize()}",
+                    client_id=f"open-zaak-{municipality}",
```
Comment (joeribekker) [inline src/openzaak/management/commands/setup_configuration.py]: lower

**Review by joeribekker — CHANGES_REQUESTED** [review]

**Review by joeribekker — APPROVED** [review]

**Pull request merged**